junit/junit.go

```diff
@@ -236,5 +237,30 @@ func formatDuration(d time.Duration) string {
 
 // formatOutput combines the lines from the given output into a single string.
 func formatOutput(output []string) string {
-	return strings.Join(output, "\n")
+	return escapeIllegalChars(strings.Join(output, "\n"))
+}
+
+func escapeIllegalChars(str string) string {
+	return strings.Map(func(r rune) rune {
+		if unicode.IsSpace(r) {
```

Comment:
Let's remove the check for `unicode.IsSpace`, because it still contains invalid characters. `isInCharacterRange` defines the entire valid range, so that's all we need here.

Reply:
How about replacing `\t` with `    ` (4 spaces). I think it's important to keep them because it makes reading easier

Reply:
Fixed